lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -190,6 +253,73 @@ void HWMemSimImplPass::runOnOperation() {
     if (genOp.descriptor() == "FIRRTL_Memory") {
       auto mem = analyzeMemOp(oldModule);
 
+      // If this memory has defined write-after-write behavior, we need to
+      // determine which sets of write ports are driven by the same clock.
+      if (mem.writeUnderWrite == 1) {
+
+        auto symbol = SymbolRefAttr::get(oldModule.getNameAttr());
+        auto instances = symbolDigger.get(symbol);
+        bool differentClocks = false;
+        auto buildWriteSets = [&](Operation *op) {
+          llvm::DenseMap<Value, unsigned> clockToLeader;
+          llvm::EquivalenceClasses<unsigned> writeSets;
+          for (size_t i = mem.numReadPorts * 3 + 0, e = op->getNumOperands();
+               i < e; i += 5) {
```

Comment:
Plz drop the "+0".  Also, 3 and 5 seem like magic numbers.  What are these?  Can/should they be enums or something else?

Reply:
There should be a better way to handle this. This is iterating over the clock operands in the generator op adhering to the `FIRRTL_Memory` schema. The schema is parametric, but organized as flat operands of `n` read ports followed by `m` write ports followed by `p` readwrite ports. The iteration over this is cleaner earlier in this file because all the ports are walked in order where they're skipped here. 

I'll see how to fix this. At minimum, just changing the loop iterator to go from 0 to `numwriteports - 1` would definitely clear up what's going on here.

Reply:
All this is now gone. 😄 This now uses actual write port indices in `LowerToHW`.